Sources/ServiceLifecycle/ServiceGroupConfiguration.swift

```diff
@@ -41,17 +44,142 @@ public struct ServiceGroupConfiguration: Hashable, Sendable {
         public init() {}
     }
 
+    public struct ServiceConfiguration: Sendable {
+        public struct TerminationBehavior: Sendable, CustomStringConvertible {
+            internal enum _TerminationBehavior {
+                case cancelGroup
+                case gracefullyShutdownGroup
+                case ignore
+            }
+
+            internal let behavior: _TerminationBehavior
+
+            public static let cancelGroup = Self.init(behavior: .cancelGroup)
+            public static let gracefullyShutdownGroup = Self.init(behavior: .gracefullyShutdownGroup)
+            public static let ignore = Self.init(behavior: .ignore)
+
+            public var description: String {
+                switch self.behavior {
+                case .cancelGroup:
+                    return "cancelGroup"
+                case .gracefullyShutdownGroup:
+                    return "gracefullyShutdownGroup"
+                case .ignore:
+                    return "ignore"
+                }
+            }
+        }
+
+        /// The service.
+        public var service: any Service
+        /// The behavior when the service returns from its `run()` method.
+        public var returnBehaviour: TerminationBehavior
+        /// The behavior when the service throws from its `run()` method.
+        public var throwBehaviour: TerminationBehavior
```

Comment:
You need to use the US spelling for the properties here.

Reply:
The naming here doesn't quite sit right with me; it's very strongly tied to language keywords rather than the behaviour of the service runner.

Not sure exactly what to suggest but the idea of a 'clean exit' vs. 'unclean exit' might be worth exploring.

Reply:
Adopted the US spelling. I am really struggling to come up with something better.

Reply:
Moved to use `successfulTerminationBehavior` and `failureTerminationBehavior`